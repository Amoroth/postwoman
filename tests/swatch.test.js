import { shallowMount } from '@vue/test-utils'
import Swatch from '@/components/settings/swatch.vue'

let wrapper

beforeEach(() => {
  wrapper = shallowMount(Swatch, { propsData: { color: 'green' } })
})

describe('Swatch Component', () => {
  it('exists', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('renders correctly', () => {
    wrapper.setProps({ name: 'example', active: true })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('hides svg if not active', () => {
    expect(wrapper.find('svg').exists()).toBeFalsy()
  })
  it('shows svg if active', () => {
    wrapper.setProps({ active: true })
    expect(wrapper.find('svg').exists()).toBeTruthy()
  })
  it('displays color if no name provided', () => {
    const div = wrapper.find('.color')
    expect(div.text()).toBe('green')
  })
  it('displays name if prop is provided', () => {
    wrapper.setProps({ name: 'hi' })
    const div = wrapper.find('.color')
    expect(div.text()).toBe('hi')
  })
})

afterEach(() => {
  wrapper.destroy()
})
