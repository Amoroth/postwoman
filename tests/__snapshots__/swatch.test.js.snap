// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swatch Component renders correctly 1`] = `
<div
  class="color"
  data-color="green"
>
  <span
    class="preview"
    style="background-color: green;"
  >
    <svg
      class="activeTick"
      clip-rule="evenodd"
      fill-rule="evenodd"
      height="24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 6.285l-11.16 12.733-6.84-6.018 1.319-1.49 5.341 4.686 9.865-11.196 1.475 1.285z"
      />
    </svg>
  </span>
  
    example

</div>
`;
